main.go

```diff
@@ -0,0 +1,240 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/events"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/urfave/cli"
+)
+
+const (
+	imageListPutImagesPath = "/images"
+)
+
+var (
+	tagRegexp = regexp.MustCompile(`:([\w][\w.-]{0,127})$`)
+	logger    = log.New(os.Stderr, "", log.Ldate|log.Ltime|log.Lshortfile)
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "imagelist-docker-events"
+	app.Version = "v0.0.1"
+	app.Usage = "poll docker for image push events and add them to imagelist service"
+
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "imagelist-url",
+			Usage: "imagelist service url",
+		},
+	}
+
+	app.Action = func(c *cli.Context) error {
+		f := filters.NewArgs()
+		f.Add("type", events.ImageEventType)
+
+		if !c.IsSet("imagelist-url") {
+			return cli.NewExitError("error: imagelist-url needs to be set", 1)
+		}
+		u := c.String("imagelist-url")
+		imageListURL, err := joinURL(u, imageListPutImagesPath)
+		if err != nil {
+			return cli.NewExitError(fmt.Sprintf("error: failed to parse imagelist url: %v", err), 1)
+		}
+
+		// main loop
+		for {
+			c, err := client.NewEnvClient()
+			if err != nil {
+				logger.Printf("error: docker connection failed: %v", err)
+			}
+			defer c.Close()
+
+			messages, errs := c.Events(context.Background(), types.EventsOptions{Filters: f})
+			if err := processEvents(messages, errs, imageListURL); err != nil {
```

Comment:
Doesn't make sense to pass `errs` into the `processEvents` function and have it handle them – maybe the `errs` can be handled _before_ calling `processEvents`?